db/schema.rb

```diff
@@ -622,18 +622,38 @@
   add_index "secure_rooms_card_readers", ["tablet_token"], name: "index_secure_rooms_card_readers_on_tablet_token", unique: true, using: :btree
 
   create_table "secure_rooms_events", force: :cascade do |t|
-    t.integer  "card_reader_id",  limit: 4
-    t.integer  "user_id",         limit: 4
+    t.integer  "card_reader_id",  limit: 4,   null: false
+    t.integer  "user_id",         limit: 4,   null: false
     t.datetime "occurred_at"
     t.string   "outcome",         limit: 255
     t.string   "outcome_details", limit: 255
     t.datetime "created_at",                  null: false
     t.datetime "updated_at",                  null: false
+    t.integer  "account_id",      limit: 4
   end
 
   add_index "secure_rooms_events", ["card_reader_id"], name: "index_secure_rooms_events_on_card_reader_id", using: :btree
   add_index "secure_rooms_events", ["user_id"], name: "index_secure_rooms_events_on_user_id", using: :btree
 
+  create_table "secure_rooms_occupancies", force: :cascade do |t|
+    t.integer  "product_id",     limit: 4, null: false
+    t.integer  "user_id",        limit: 4, null: false
+    t.integer  "account_id",     limit: 4
+    t.integer  "entry_event_id", limit: 4
+    t.datetime "entry_at"
+    t.integer  "exit_event_id",  limit: 4
+    t.datetime "exit_at"
+    t.datetime "orphan"
```

Comment:
Would this be better named `orphaned_at`? Making a method `orphan?` would probably be worth adding.